yt_dlp/extractor/redtube.py

```diff
@@ -84,24 +87,33 @@ def _real_extract(self, url):
                 r'mediaDefinition["\']?\s*:\s*(\[.+?}\s*\])', webpage,
                 'media definitions', default='{}'),
             video_id, fatal=False)
-        if medias and isinstance(medias, list):
-            for media in medias:
+        for media in medias if isinstance(medias, list) else []:
+            format_url = url_or_none(media.get('videoUrl'))
+            if not format_url:
+                continue
+            format_id = media.get('format')
+            quality = media.get('quality')
+            if format_id == 'hls' or (format_id == 'mp4' and not quality):
+                more_media = self._download_json(format_url, video_id, fatal=False)
+            else:
+                more_media = [media]
+            for media in more_media if isinstance(more_media, list) else []:
```

Comment:
any reason for the isinstance check? Cant this just be `more_media or []`? Same above

Reply:
I guess is a copy of line 90...

Reply:
If `more_media` was parsed from JSON (a value returned by `json.loads()`), we don't know that it's iterable: it might be `True`, or `42`.